fechar duas activities ao mesmo tempo para voltar a MainActivity

rubensfelipe opened the pull request:
… uma foto na tela "nova receita foto" e ao clicar em atualizar na tela "editar receita", a intent que levava o usuário para MainActivity foi substituída por um método que encerra a activity "nova receita info" e a activity "nova receita foto" ao mesmo tempo. Mesmo procedimento foi adotado para tela editar receita. Ao clicar em atualizar as activities "editar receita" e "visualizar receita" são encerradas. Através do método finish.

Resolvido o problema de activities antigas e fora de ordem aparecerem após o usuário clicar no botão voltar do celular.

rubensfelipe committed:
na tela de nova receita foto e na tela de editar receita, ao escolher uma foto na tela "nova receita foto" e ao clicar em atualizar na tela "editar receita", a intent que levava o usuário para MainActivity foi substituída por um método que encerra a activity "nova receita info" e a activity "nova receita foto" ao mesmo tempo. Mesmo procedimento foi adotado para tela editar receita. Ao clicar em atualizar as activities "editar receita" e "visualizar receita" são encerradas. Através do método finish.

Resolvido o problema de activities antigas e fora de ordem aparecerem após o usuário clicar no botão voltar do celular.

rubensfelipe commented:
na tela de nova receita foto e na tela de editar receita, ao escolher uma foto na tela "nova receita foto" e ao clicar em atualizar na tela "editar receita", a intent que levava o usuário para MainActivity foi substituída por um método que encerra a activity "nova receita info" e a activity "nova receita foto" ao mesmo tempo. Mesmo procedimento foi adotado para tela editar receita. Ao clicar em atualizar as activities "editar receita" e "visualizar receita" são encerradas. Através do método finish.

Resolvido o problema de activities antigas e fora de ordem aparecerem após o usuário clicar no botão voltar do celular.

RodrigoDavy commented on app/src/main/java/com/example/android/receitaboa/activity/MainActivity.java:
```diff
@@ -54,6 +54,11 @@ protected void onCreate(Bundle savedInstanceState) {
         viewPagerTab.setViewPager(viewPager);
 
+        if (getIntent().getBooleanExtra("FECHAR_ACTIVITY",false)){
+            Toast.makeText(getApplicationContext(),"FECHANDO",Toast.LENGTH_SHORT).show();
```
Extrair FECHANDO para string.xml

rubensfelipe replied:
desisti desse método. Vou remove-lo.

RodrigoDavy commented on app/src/main/java/com/example/android/receitaboa/activity/NovaReceitaFotoActivity.java:
```diff
@@ -141,11 +141,12 @@ public void onClick(DialogInterface dialog, int which) {
                 Toast.makeText(getApplicationContext(), getApplicationContext().getString(R.string.nova_receita_adicionada), Toast.LENGTH_SHORT).show();
+
+                //encerra a activity anterior ao salvar a foto da receita
+                NovaReceitaInfoActivity.addInfoReceita.finish();
+
+                finish();
```
Código repetido, isolar em uma função única

Lembrar sempre do princípio DRY (Don't Repeat Yourse): https://dzone.com/articles/software-design-principles-dry-and-kiss

RodrigoDavy commented on app/src/main/java/com/example/android/receitaboa/activity/NovaReceitaFotoActivity.java:
```diff
@@ -273,11 +269,13 @@ public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                     mensagemFotoAdicionada();
+
+                    //encerra a activity anterior ao salvar a foto da receita
+                    NovaReceitaInfoActivity.addInfoReceita.finish();
+
+                    finish();
```
Código repetido, isolar em uma função única

Lembrar sempre do princípio DRY (Don't Repeat Yourse): https://dzone.com/articles/software-design-principles-dry-and-kiss

rubensfelipe committed:
demandas do PR solucionadas.

Além disso, mudanças nas telas Nova Receita Info e Nova Receita Foto.

novo método para recuperar o id da receita gerada na tela Nova Receita Info.

após clicar no botão salvar na tela (Nova Receita Info):
Método intent.putExtra(novaReceita.getId)

na tela (Nova Receita Foto)
getExtra("idReceita") //recuperando id da receita gerada na tela anterior

RodrigoDavy commented on app/src/main/java/com/example/android/receitaboa/activity/EditarReceitaActivity.java:
```diff
@@ -157,8 +158,8 @@ public void onClick(View v) {
+                //encerra a activity anterior ao clicar no botão atualizar
+                VisualizarReceitaActivity.visualizacaoReceita.finish();
 
                 finish();
```
Isso tá fazendo essencialmente a mesma coisa que o fecharActivityAtualAnterior(), a única diferença é que a atividade anterior muda. Isso não deixa de ser código repetido

Poderia mudar o método fecharActivityAtualAnterior para receber como parâmetro a activity anterior a ser fechada e aplicar o finish() nela

RodrigoDavy commented on app/src/main/java/com/example/android/receitaboa/activity/VisualizarReceitaActivity.java:
```diff
@@ -39,6 +40,7 @@
     private DatabaseReference receitasRef;
     private DatabaseReference receitasChefRef;
 
+    public static Activity visualizacaoReceita;
```
mudar para um nome mais genérico

RodrigoDavy commented on app/src/main/java/com/example/android/receitaboa/activity/NovaReceitaInfoActivity.java:
```diff
@@ -45,12 +43,17 @@
     private StorageReference storageRef;
     private String identificadorChef;
 
+    public static Activity addInfoReceita;
```
mudar para um nome mais genérico

rubensfelipe committed:
ajuste das funções
fecharAtividadeAtualAnterior(Activity atividadeAnterior);
